src/Core/Geometry/TriangleMesh.hpp

```diff
@@ -249,6 +250,13 @@ class IndexedGeometry : public AttribArrayGeometry
     /// \warning There is no error check on the handles attribute type.
     inline bool append( const IndexedGeometry<IndexType>& other );
 
+    const IndexContainerType& getIndices() const { return m_indices; }
+    /// not allowed to have notification
+    /// IndexContainerType& getIndices() { return m_indices; }
+    void setIndices( const IndexContainerType indices ) {
+        m_indices = indices;
+        notify();
+    }
     ///\todo make it protected
```

Comment:
protected data might prevent notifiction and invariant maintenance.
https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#Rh-protected
prefer make it private